src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -122,7 +122,7 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                     if (addresses.SingleOrDefault() != "http://localhost:5000")
                     {
                         var joined = string.Join(", ", addresses);
-                        throw new NotSupportedException($"Specifying address(es) '{joined}' is incompatible with also configuring endpoint(s) in UseKestrel.");
+                        _logger.LogWarning($"Address(es) '{joined}' were overridden. Binding to endpoints defined in UseKestrel instead.");
```

Comment:
Referencing UseUrls() or IServerAddressesFeature is the tricky part. UseUrls() might not be the thing that configured the addresses, and I doubt most ASP.NET Core devs know or care what IServerAddressesFeature is. That's why I just show the addresses that were configured.